### PR TITLE
[improve][broker] Failed to start broker when Linux load balancer can't get NIC speed

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

The context here: #14537

The current design is frequently logging an error while ignoring the NIC. The only real course of action for an operator is to reconfigure the broker. It seems like a better course of action to fail on startup so that operators know immediately that they need to fix the configuration instead of finding out sometime later when observing the logs.

### Modifications

- Verify whether VM has NIC speed.
- Refactor ``LinuxBrokerHostUsageImpl`` and extract some Linux operate to another class.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

**Comment by gaozhangmin** [comment]
linux host has many NICs, There are some NIC that are of type 1 that doesn't expose the speed. 

 I think if let broker fail it's not  a good idea. @mattisonchao

**Comment by gaozhangmin** [comment]
linux host has many NICs, There are some NIC that are of type 1 that doesn't expose the speed. 

 I think if let broker fail it's not  a good idea. @mattisonchao

**Comment by mattisonchao** [comment]
@gaozhangmin 

Sorry, I make a mistake.
We need to get all NIC speeds and then check them, if we don't get any NIC speeds (less than or equal to 0), we need to notify the user to set "loadBalancerOverrideBrokerNicSpeedGbps" to make sure the load balancer works.

**Comment by mattisonchao** [comment]
@gaozhangmin 

If the Linux host has a legitimate NIC but doesn't have the speed. Regardless of how many NICs have speed, we should notify the user to set ``loadBalancerOverrideBrokerNicSpeedGbps`` to ensure that the user gets what they expect.

**Comment by mattisonchao** [comment]
@nicoloboschi  @eolivelli @michaeljmarshall  please continue to take a look from #14537.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 1]
```diff
@@ -635,6 +636,13 @@ public void start() throws PulsarServerException {
                         + "authenticationEnabled=true when authorization is enabled with authorizationEnabled=true.");
             }
 
+            if (config.isLoadBalancerEnabled() && LinuxInfoUtils.isLinux()) {
```
Comment (nicoloboschi) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:639]: If I set `loadBalancerOverrideBrokerNicSpeedGbps` I should not get the error, right?

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
Overall LGTM, but I have few questions:
1. We should add unit tests for checking the logic
2. What if I run the Pulsar tests within a machina with NIC without 'speed' file? The test will fail ? 
For instance, I've seen this kind of NIC inside a ubuntu VM with docker installed

**Comment by Nicklee007** [comment]
> @gaozhangmin
> 
> If the Linux host has a legitimate NIC but doesn't have the speed. Regardless of how many NICs have speed, we should notify the user to set `loadBalancerOverrideBrokerNicSpeedGbps` to ensure that the user gets what they expect.

@mattisonchao 
I think  that we need check all NIC's speed configuration is too strict to start server. In some case we only used eth0 and the eth1 eth2  ... ... not active, but those /sys/class/net/eth1/type configured '1' and read 'speed' file is 'Invalid argument';
Maybe those not active NIC need excluded from the NICs list which is checked 'speed' file and calculated speed.

**Comment by mattisonchao** [comment]
Hi, @Nicklee007  @gaozhangmin 

I changed the logic to fail to start the broker when we can't get the NIC speed (<= 0).

**Comment by mattisonchao** [comment]
Hi, @nicoloboschi 
I have already added tests for this logic.

**Review by Jason918 — APPROVED** [review]
Overall LGTM

**Review by codelipenghui — APPROVED** [review]

**Comment by mattisonchao** [comment]
@nicoloboschi  @gaozhangmin  PTAL:)

**Review by gaozhangmin — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
Great work

**Comment by ZiyaoWei** [comment]
I'm trying to set up a dev environment using an AWS EC2 instance (tired of fighting my M1 MBP 🥲), but (as @nicoloboschi mentioned) the tests are failing due to not having NIC speed information:

```
ubuntu@ip-172-31-68-254:~/pulsar$ cat /sys/class/net/eth0/speed
cat: /sys/class/net/eth0/speed: Invalid argument
```

I opened #15263 trying to fix the tests but there are way too many of them - I prefer having the tests not depending on external factors, so maybe we should either fix the tests (which isn't too practical since people would add new tests without specifying NIC speed information) or somehow let the tests skip this check?

**Comment by lhotari** [comment]
This broke unit tests on Linux when there's no ethernet connection. I fixed the problem for all unit tests in #15365 .

**Comment by duxugang** [comment]
I ran into the same problem with version 2.10

**Comment by mattisonchao** [comment]
@duxugang  Hi, this PR does not cherry-pick to branch-2.10

**Pull request merged**